hs_core/hydroshare/utils.py

```diff
@@ -433,11 +433,19 @@ def copy_resource_files_and_AVUs(src_res_id, dest_res_id, set_to_private=False):
     tgt_coll = tgt_res.root_path
     for avu_name in avu_list:
         value = istorage.getAVU(src_coll, avu_name)
+        # possibly create private copy if requested
         if avu_name == 'isPublic':
             if set_to_private:
+                tgt_coll.raccess.public = False
                 istorage.setAVU(tgt_coll, avu_name, 'False')
             else:
+                tgt_coll.raccess.public = value
```

Comment:
@pkdash @hyi I have checked the code. The sequence of events that happens is always (by some mechanism): 
```
copy_or_version = hydroshare.create_empty_resource(source.short_id, new_owner)
copy_resource_files_and_AVUs(source.short_id, copy_or_version.short_id, set_to_private)
.... other stuff .... 
```
This is invariant of whether a version or copy is being created. In this sequence, 

* new owner is sole owner. 
* access control is otherwise not copied. 
* all resource flags are set to defaults
    * public = False
    * discoverable = False
    * shareable = False
...

So: 

* Before this PR, the resource is private in access control before this.  
* public -> discoverable
* Thus these flags are incorrect before this PR. 
* However, none of the fancy features of `set_resource_sharing_status` are used, either. 

My recommendation is not to use `set_resource_sharing_status` for this, but instead simply set both `discoverable=True` and `public=True` if there is a source with `isPublic=True` and not `set_to_private`.

Long term, this needs some refactoring. `create_empty_resource` doesn't create an empty resource, it creates a copy/version without files, but based upon an empty resource. A better name is `clone_resource_without_files`. On a deeper level, this should be private. It is only used by copy and version commands. Yes, I know, it is tested as a public function. 
 
So, long term, I recommend: 
* call `create_empty_resource` inside `create_resource` and `create_new_version_resource`. 
* Make the `create_empty_resource` method private. 
* Change to a more descriptive name.